shoplist/src/main/java/com/netguru/codereview/ui/MainFragment.kt

```diff
@@ -10,18 +10,14 @@ import android.widget.ProgressBar
 import android.widget.Toast
 import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.ViewModelProvider
+import androidx.fragment.app.viewModels
 import coil.load
-import com.netguru.codereview.network.model.ShopListItemResponse
-import com.netguru.codereview.network.model.ShopListResponse
 import com.netguru.codereview.shoplist.R
-import com.netguru.codereview.ui.model.ShopList
-import javax.inject.Inject
+import com.netguru.codereview.ui.model.ResultState
 
 class MainFragment : Fragment() {
 
-    @Inject
-    private var viewModel: MainViewModel? = null
+    private val viewModel: MainViewModel by viewModels()
```

Comment:
use Kotlin Property delegation syntax, instead of repeating android more common pattern, when declaring ViewModel
, https://developer.android.com/kotlin/common-patterns#delegate